tests/e2e/Services/Migrations/MigrationsBase.php

```diff
@@ -896,4 +898,228 @@ public function testAppwriteMigrationFunction(): void
+                'bucketId' => $bucketIds['uncompressed'],
+                'resourceId' => $databaseId . ':' . $collectionId,
+            ]
+        );
+
+        $this->assertEmpty($migration['body']['statusCounters']);
+        $this->assertEquals('CSV', $migration['body']['source']);
+        $this->assertEquals('pending', $migration['body']['status']);
+        $this->assertEquals('Appwrite', $migration['body']['destination']);
+        $this->assertContains(Resource::TYPE_DOCUMENT, $migration['body']['resources']);
+
+        return [
+            'databaseId' => $databaseId,
+            'collectionId' => $collectionId,
+            'migrationId' => $migration['body']['$id'],
+        ];
+    }
+
+    /**
+     * @depends testCreateCsvMigration
+     */
+    public function testImportSuccessful(array $response): void
+    {
+        $databaseId = $response['databaseId'];
+        $collectionId = $response['collectionId'];
+        $migrationId = $response['migrationId'];
+
+        $documentsCountInCSV = 100;
+
+        // get migration stats
+        $this->assertEventually(function () use ($migrationId, $databaseId, $collectionId, $documentsCountInCSV) {
+            $migration = $this->client->call(Client::METHOD_GET, '/migrations/'.$migrationId, array_merge([
+                'content-type' => 'application/json',
+                'x-appwrite-project' => $this->getProject()['$id'],
+            ], $this->getHeaders()));
+
+            $this->assertEquals(200, $migration['headers']['status-code']);
+            $this->assertEquals('finished', $migration['body']['stage']);
+            $this->assertEquals('completed', $migration['body']['status']);
+            $this->assertEquals('CSV', $migration['body']['source']);
+            $this->assertEquals('Appwrite', $migration['body']['destination']);
+            $this->assertContains(Resource::TYPE_DOCUMENT, $migration['body']['resources']);
+            $this->assertArrayHasKey(Resource::TYPE_DOCUMENT, $migration['body']['statusCounters']);
+            $this->assertEquals($documentsCountInCSV, $migration['body']['statusCounters'][Resource::TYPE_DOCUMENT]['success']);
+        }, 60000, 500);
+
+        // get documents count
+        $documents = $this->client->call(Client::METHOD_GET, '/databases/'.$databaseId.'/collections/'.$collectionId.'/documents', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'queries' => [
+                // there should be only 100!
+                Query::limit(150)->toString()
+            ]
+        ]);
+
+        $this->assertEquals(200, $documents['headers']['status-code']);
+        $this->assertIsArray($documents['body']['documents']);
+        $this->assertIsNumeric($documents['body']['total']);
+        $this->assertEquals($documentsCountInCSV, $documents['body']['total']);
+    }
+
```

Comment:
Let's check more failure cases: 
- Invalid attribute in header row
- Missing required attribute in header row
- Missing attribute in document row
- Invalid attribute in document row

The first two should throw a 400 from the API. The other two should fail the job in the worker

Reply:
All need to come from the Worker.
Added the tests with respect to these changes - https://github.com/utopia-php/migration/pull/75!